drivers/hmis/app/graphql/types/admin/form_item_input.rb

```diff
@@ -0,0 +1,43 @@
+#  Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+#  License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+#
+
+module Types
+  # This is the GraphQL input version of Types::Forms::FormItem. This input type is needed because
+  # in graphql, attributes on input types can only be built-in scalar types or enums.
+  class Admin::FormItemInput < Types::BaseInputObject
+    argument :link_id, String, 'Unique identifier for item', required: false
```

Comment:
Ahh sigh, when we discussed this approach earlier, I didn't consider the fact that we can't actually re-use `Types::Forms::FormItem` as the input type. Part of the benefit, in my mind, is that we could use the existing type. But you're right that's [not possible](https://spec.graphql.org/October2021/#note-7a6d2).


It feels redundant to have the Form structure defined in three places:
1) The [JSON Schema](https://github.com/greenriver/hmis-warehouse/blob/release-118/drivers/hmis_external_apis/public/schemas/form_definition.json) which is used for validating the form structure
2) `Types::Forms::FormItem` which is used for resolving the form
3) `Types::Admin::FormItemInput` which is used for updating the form


I'm thinking that for _updating the form_, we already rely on the JSON Schema (1) to validate the input, when we call validate_schema. Is it really necessary to ALSO have the GraphQL layer validate the input shape (3)?

The old mutation (`UpdateFormDefinition`) just accepts the form as a string, and then validates it using the JSON Schema. I feel like we COULD stick with that approach, the only difference we need is for the mutation to `.underscore` all the keys in the json string that it receives, prior to validating and saving it.

That would let us: (1) continue to rely on the JSON Schema for validation, and (2) let the frontend continue to use the GraphQL Typed (camelCased) form shape. The frontend just needs to stringify it before sending it to the mutation. Is that a viable path, or am I missing something? Sorry for the waffling here

Reply:
If that's not possible, I also looked into generating the `Admin::FormItemInput` class which seems very doable

Reply:
Ok, I took a stab at implementing your suggestion. It is a bit more complicated than simply calling .underscore, because of having to remove junk like `__typename` (which previously I was doing on the frontend here: https://github.com/greenriver/hmis-frontend/pull/785/commits/bb35bc177bf36507e53bb77a7dfec92b339eba2f#diff-7d33d3e9a02ac3ebc1f517b289868d09de568774095409317f30b95d4c1d1341R278-R306)

I'd be ok to keep doing that on the frontend, if you prefer the backend should never know about that cruft, but I think I prefer it this way where only one or the other (backend or frontend) have to implement this recursive validation. What do you think?